### PR TITLE
Proofreading language-basics.adoc through line 1980.

### PR DESCRIPTION
**Pull request by jdeisenberg**

### CODE REVIEW

File: src/language-basics.adoc [hunk 1]
```diff
@@ -1099,8 +1099,11 @@ TODO: think about this.
 
 ===== nil-punning
 
+Since `seq` returns `nil` when the collection is empty, and `nil` evaluates to false in boolean context, you can check to see if a collection is empty by using the `seq` function. The technical term for this is nil-punning.
+////
 The above behaviour of `seq` when coupled with the falsey nature of `nil` in boolean contexts make it an idiom for checking
```
Comment (dialelo) [inline src/language-basics.adoc]: Way better explanation :+1:

File: src/language-basics.adoc [hunk 2]
```diff
@@ -1303,6 +1316,33 @@ treating potentially infinite sequences as regular ones.
...
+;; => (0 1 2 3 4)
+(range 1 10)
+;; => (1 2 3 4 5 6 7 8 9)
+(range 10 100 15)
+;; (10 25 40 55 70 85)
+---
+
+If you just say `(range)`, you will get an infinite sequence of all the integers.
+Do *not* try this in the REPL, unless you are prepared to wait for a very, very long time, because
+the REPL wants to fully evaulate the expression.
```
Comment (dialelo) [inline src/language-basics.adoc]: There is a typo: evaulate instead of evaluate.

File: src/language-basics.adoc [hunk 3]
```diff
@@ -1391,10 +1431,7 @@ another collection with `vec`:
 ;; => true
 
 (vector 0 1 2)
-;; => [0 1 2]
-
-(vec '(0 1 2))
-;; => [0 1 2]
+;; => [0 1[0 1 2]
```
Comment (dialelo) [inline src/language-basics.adoc]: Looks like some code got deleted accidentaly here.

File: src/language-basics.adoc [hunk 4]
```diff
@@ -1496,11 +1536,13 @@ faster than building a vector from a lazy sequence and making your intent more e
...
 Maps are ubiquitous in ClojureScript. Like vectors, they are also used as a syntactic construct for attaching metadata to
-vars. Any ClojureScript data structure can be used as a key in a map, although it's common to use keywords since they can
+vars. 
+Any ClojureScript data structure can be used as a key in a map, although it's common to use keywords since they can
 also be called as functions.
 
 ////
 Joe the Beginning Programmer asks: "What do you mean by 'attaching metadata to vars'? I'm confused."
+Really, you need a reference to the section where you discuss metadata, or you need to get rid of it in the preceding text.
 ////
```
Comment (dialelo) [inline src/language-basics.adoc]: I'd go with a reference to the metadata section.

**Comment by dialelo** [comment]
Thanks for your effort @jdeisenberg, you're making the book a lot better. I left some comments in the PR, let me know what you think.

File: src/language-basics.adoc (hunk 2 above)
Comment (jdeisenberg) [inline src/language-basics.adoc]: I'll fix it in my next PR.

File: src/language-basics.adoc (hunk 3 above)
Comment (jdeisenberg) [inline src/language-basics.adoc]: Yeah, I think that was when Jasper the cat jumped up on the desk.

**Comment by dialelo** [comment]
Since there were merge conflicts I solved them locally and pushed these changes to the `master` branch. Excellent work, thanks a lot!

**Pull request closed without merging**